File: app/src/main/java/com/jajinba/pixabaydemo/contract/ListContract.java
package com.jajinba.pixabaydemo.contract;


import com.jajinba.pixabaydemo.model.PixabayImageObject;

import java.util.List;

import io.reactivex.annotations.Nullable;

import static com.jajinba.pixabaydemo.model.ImageManager.Operation;

public interface ListContract {

  interface View {
    void searchFinished(@Nullable String keyword, @Nullable List<PixabayImageObject> imageList);
  }

  interface Presenter {
    List<PixabayImageObject> getImageList(String keyword);

    void loadMore(String keyword);

    @Operation String getLastOperation();
  }

}

File: app/src/main/java/com/jajinba/pixabaydemo/contract/ListContract.kt
package com.jajinba.pixabaydemo.contract

import androidx.annotation.StringRes
import com.jajinba.pixabaydemo.model.ImageManager
import com.jajinba.pixabaydemo.model.PixabayImageObject

interface ListContract {
    interface View {
        fun searchFinished(keyword: String, imageList: MutableList<PixabayImageObject>)
        fun onFailed(@StringRes errorMsg: Int)
    }

    interface Presenter {
        fun getImageList(keyword: String): MutableList<PixabayImageObject>
        fun loadMore(keyword: String)

        @ImageManager.Operation
        fun getLastOperation(): String
    }
}
